pkg/machine/hyperv/stubber.go

```diff
@@ -0,0 +1,577 @@
+//go:build windows
+
+package hyperv
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/Microsoft/go-winio"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	gvproxy "github.com/containers/gvisor-tap-vsock/pkg/types"
+	"github.com/containers/libhvee/pkg/hypervctl"
+	"github.com/containers/podman/v4/pkg/machine"
+	"github.com/containers/podman/v4/pkg/machine/define"
+	"github.com/containers/podman/v4/pkg/machine/hyperv/vsock"
+	"github.com/containers/podman/v4/pkg/machine/ignition"
+	"github.com/containers/podman/v4/pkg/machine/vmconfigs"
+	"github.com/containers/podman/v4/pkg/strongunits"
+	"github.com/containers/podman/v4/pkg/systemd/parser"
+	"github.com/sirupsen/logrus"
+)
+
+type HyperVStubber struct {
+	vmconfigs.HyperVConfig
+}
+
+func (h HyperVStubber) CreateVM(opts define.CreateVMOpts, mc *vmconfigs.MachineConfig, builder *ignition.IgnitionBuilder) error {
+	var (
+		err error
+	)
+	callbackFuncs := machine.InitCleanup()
+	defer callbackFuncs.CleanIfErr(&err)
+	go callbackFuncs.CleanOnSignal()
+
+	hwConfig := hypervctl.HardwareConfig{
+		CPUs:     uint16(mc.Resources.CPUs),
+		DiskPath: mc.ImagePath.GetPath(),
+		DiskSize: mc.Resources.DiskSize,
+		Memory:   mc.Resources.Memory,
+	}
+
+	networkHVSock, err := vsock.NewHVSockRegistryEntry(mc.Name, vsock.Network)
+	if err != nil {
+		return err
+	}
+
+	mc.HyperVHypervisor.NetworkVSock = *networkHVSock
+
+	// Add vsock port numbers to mounts
+	err = createShares(mc)
+	if err != nil {
+		return err
+	}
+
+	removeShareCallBack := func() error {
+		return removeShares(mc)
+	}
+	callbackFuncs.Add(removeShareCallBack)
+
+	if err := addNetworkAndReadySocketsToRegistry(mc); err != nil {
+		return err
+	}
```

Comment:
This doesn't really do anything. We already assign a network socket, which we then overwrite, and then the ready socket code is commented out.